nodes/generator/torusKnot.py

```diff
@@ -320,9 +343,9 @@ def sv_init(self, context):
 
         self.outputs.new('VerticesSocket', "Vertices")
         self.outputs.new('StringsSocket',  "Edges")
+        self.outputs.new('VerticesSocket', "Tangents")
```

Comment:
adding sockets inbetween existing sockets will cause .blends that use the existing sockets to output unexpected data when loaded with the new version of the Node. However, if you add the new socket as a last socket things will be OK.

also json IO works on index and not socketname.

things to consider.

Reply:
good point. I thought that regardless of what you change in a node you'll have to update those node-trees using an older version of the node anyway.. so if the tanget output i added at the end of within other sockets it wouldn't have mattered (it made more sense to me for the tangent to come before normal). 

And btw..
1. It would be nice if you could reorder sockets and have the node still work correctly (not sure if that is possible). 
2. Sometimes I get this nodes with a red box around them hinting that the node is obsolete / outdated. But I don't get the same for the torus knot node once i updated it to a new version. Any ideas why? 
3. The new() for sockets take a 3rd argument which i see in many nodes being populated with the same as the second argument. Any reasons for that? I thought that one is supposed to be the name of the socket, and the other the socket label displayed in the UI. If that would be the case you could easily rename the socket LABEL in one place and not have to update all the code referencing the NAME of that socket.